rules.go

```diff
@@ -0,0 +1,44 @@
+package shopify
+
+// Rules is a list of rules
+type Rules []Rule
+
+// Rule define what products go into the smart collection.
+type Rule struct {
+	// column is the property of a product being used to populate the smart collection.
+	/* Valid values for text relations:
+	   	- title: The product title.
+	   	- type: The product type.
+	   	- vendor: The name of the product vendor.
+	   	- variant_title: The title of a product variant.
+	   Valid values for number relations:
+	   	- variant_compare_at_price: The compare price.
+	   	- variant_weight: The weight of the product.
+	   	- variant_inventory: The inventory stock. Note: not_equals does not work with this property.
+	   	- variant_price: product price.
+	   Valid values for an equals relation:
+	   	- tag: A tag associated with the product.
+	*/
+	Column string
+	// Relation is the relationship between the column choice, and the condition.
+	/*
+		Valid values for number relations:
+		- greater_than: The column value is greater than the condition.
+		- less_than: The column value is less than the condition.
+		- equals: The column value is equal to the condition.
+		- not_equals: The column value is not equal to the condition.
+		Valid values for text relations:
+		- equals: Checks if the column value is equal to the condition value.
+		- not_equals: Checks if the column value is not equal to the condition value.
+		- starts_with: Checks if the column value starts with the condition value.
+		- ends_with: Checks if the column value ends with the condition value.
+		- contains: Checks if the column value contains the condition value.
+		- not_contains: Checks if the column value does not contain the condition value.
+	*/
+	Relation string
+	// Condition is the select products for a smart collection using a condition.
+	/*
+		Values are either strings or numbers, depending on the relation value.
```

Comment:
get rid of this comment as it can't be true in go, we'll cast any ints into strings